sklearn/manifold/_t_sne.py

```diff
@@ -666,6 +682,15 @@ def _fit(self, X, skip_num_points=0):
             raise ValueError("'method' must be 'barnes_hut' or 'exact'")
         if self.angle < 0.0 or self.angle > 1.0:
             raise ValueError("'angle' must be between 0.0 - 1.0")
+        if self.square_distances not in [True, 'legacy']:
+            raise ValueError("'square_distances' must be True or 'legacy'.")
+        if self.metric != "euclidean" and self.square_distances is not True:
+            warnings.warn(("'square_distances' has been introduced in 0.24. "
+                           "'legacy' is provided as a setting for backward "
+                           "compatibility purposes. However, 'legacy' will "
+                           "be removed in 0.26, at which point all distance "
+                           "metrics will be squared by default."),
```

Comment:
let's add

> Set square_distances=True to avoid this warning. Note that this parameter will be removed in Y.

Reply:
Hmm. Do you have any thoughts on https://github.com/scikit-learn/scikit-learn/pull/17662#discussion_r450137420?

Reply:
I'd still add it, because otherwise users have no clue what to do unless they go check out the new docs.

Reply:
Thanks! I've re-written the message in a way that I hope make things clearer. :)